haystack_experimental/components/agents/human_in_the_loop/strategies.py

```diff
@@ -107,3 +130,290 @@ def from_dict(cls, data: dict[str, Any]) -> "HumanInTheLoopStrategy":
         if not hasattr(ui_class, "from_dict"):
             raise ValueError(f"Class {ui_class} does not implement from_dict method.")
         return cls(confirmation_policy=policy_class.from_dict(policy_data), confirmation_ui=ui_class.from_dict(ui_data))
+
+
+class BreakpointConfirmationStrategy:
+    """
+    Confirmation strategy that raises a tool breakpoint exception to pause execution and gather user feedback.
+
+    This strategy is designed for scenarios where immediate user interaction is not possible, such as in backend
+    services. When a tool execution requires confirmation, it raises an `ToolBreakpointException`, which is caught
+    by the Agent. The Agent can then serialize its current state, including the tool call details, and send this
+    information to a front-end interface for user review.
```

Comment:
This is the new strategy that raises a `ToolBreakpointException` to stop Agent execution and save a snapshot.